package/MDAnalysis/auxiliary/XVG.py

```diff
@@ -0,0 +1,245 @@
+   :members:
+
+"""
+
+import os
+import numpy as np
+from . import base
+
+def uncomment(lines):
+    """ Remove comments from lines in an .xvg file
+
+    Parameters
+    ----------
+    lines : list of str
+        Lines as directly read from .xvg file
+
+    Yields
+    ------
+    str
+        The next non-comment line, with any trailing comments removed
+    """
+    for line in lines:
+        # '@' must be at the beginning of a line to be a grace instruction
+        if line.lstrip()[0] == '@':
+            continue
+        # '#' can be anywhere in the line, everything after is a comment
+        comment_position = line.find('#')
+        if comment_position > 0 and line[:comment_position]:
+            yield line[:comment_position]
+        elif comment_position < 0 and line:
+            yield line
+        # if comment_position == 0, then the line is empty
+
+
+class XVGReader(base.AuxReader):
+    """ Auxiliary reader to read data from an .xvg file.
+
+    All data from the file will be read and stored on initialisation. This is
+    the default reader for .xvg files.
+    
+    xvg files are produced by Gromacs during simulation or analysis, formatted
+    for plotting data with Grace.
+
+    Parameters
+    ----------
+    filename : str
+       Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.    
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxReader`
+
+    Note
+    ----
+    The time of each step is assumed to stored in the first column (``time_col``
+    defaults to 0), in units of ps, and data is assumed to be time-ordered.
+
+    The file is assumed to be of a size such that reading and storing the full 
+    contents is practical.
+    """
+
+    format = "XVG"
+
+    def __init__(self, filename, **kwargs):
+        self._data_input = os.path.abspath(filename)
+        with open(filename) as xvg_file:
+            lines = xvg_file.readlines()
+        auxdata = []
+        # remove comments before storing
+        for i, line in enumerate(uncomment(lines)):
+           auxdata.append([float(l) for l in line.split()])
+           # check the number of columns is consistent
+           if len(auxdata[i]) != len(auxdata[0]):
+                raise ValueError('Step {0} has {1} columns instead of '
+                                 '{2}'.format(i, auxdata[i], auxdata[0]))
+        self._auxdata = np.array(auxdata)
+        # default to time in first column, if not otherwise specified
+        time_col = kwargs.pop('time_col', 0)
+        super(XVGReader, self).__init__(time_col=time_col, **kwargs)
+
+    def _read_next_step(self):
+        """ Read next auxiliary step. """
+        new_step = self.step + 1
+        if new_step < self.n_steps:
+            self._data = self._auxdata[new_step]
+            self.step = new_step
+        else:
+            self.go_to_first_step()
+            raise StopIteration
+
+    def go_to_step(self, i):

```

Comment:
Mostly because I didn't quite understand `__getitem__` when I first started >\_< 
but yes, it's been hovering on my to-do list to make a `__getitem__` like in the trajectory reader; I can get onto that now